org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -413,7 +419,15 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
 		if ($.getTextEdit() != null) {
 			String newText = $.getTextEdit().isLeft() ? $.getTextEdit().getLeft().getNewText() : $.getTextEdit().getRight().getNewText();
 			Range range = $.getTextEdit().isLeft() ? $.getTextEdit().getLeft().getRange() : ($.getTextEdit().getRight().getInsert() != null ? $.getTextEdit().getRight().getInsert() : $.getTextEdit().getRight().getReplace());
-			if (range != null && newText != null) {
```

Comment:
I think there needs to be an additional check here for https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/ff4cfaa3bcb579749ac2194eb58562b0b93ae4b5/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalDescriptionProvider.java#L808 . The whole reason we have to adjust the `filterText` is because of `labelDetails`. We can no longer rely on `label` to ensure filtering works.

However some clients may not support `labelDetails` and they probably will continue to work with `label`. I'm not sure if this change would be harmless to them. So it's best to just confirm that we are only setting the `filterText` if we know `labelDetails` is supported.

Reply:
Per specification clients should use `filterText` if present, and `label` as fallback for filtering. And some clients only support prefix filtering, no fuzzy matching.

It doesn't mention `labelDetails` in regards to filtering - that should only be used as decoration/information so it shouldn't impact the filtering for specification conforming clients.

Reply:
Yes, but if some clients were filtering based on `label`, this would support their previous behaviour (continuing to work with `label`).

I agree that longer term, we're just going to have to support filterText everywhere because managing the subtle differences across clients isn't very sustainable.